src/error.rs

```diff
@@ -238,6 +249,35 @@ impl std::convert::From<CertError> for UserApiError {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+/// Errors which may be encountered when handling Version Loaded Endorsement Keys
+pub enum VlekError {
+    /// Certificate length does not match what was specified in the buffer.
+    InvalidLength,
+
+    /// No certificates were provided
+    EmptyCertBuffer,
+
+    /// Unknown Error.
+    UnknownError,
+}
+
```

Comment:
Why create a VLEK error instead of using the existing Cert error?

Reply:
When the platform owner calls SNP_VLEK_LOAD, they are passing a Wrapped Hashstick, not the x509 certificate itself. The certificate is provided to the guest via the extended guest request workflow.

Reply:
These errors are separate, as they are not certificate errors per se.

Reply:
Do you think maybe we refer to them has `HashstickErrors`, instead?